src/app

```diff
@@ -52,43 +52,29 @@ if test -d "$PERSISTENT_STORE"; then
     fi
 fi
 
-CONFIG_HAS_XDELTA3_CMD="xdelta3"
-CONFIG_HAS_TAR_CMD="tar"
-delta_cmd="${CONFIG_HAS_XDELTA3_CMD} -d -s"
-tar_decompress_cmd="${CONFIG_HAS_TAR_CMD} -xzvf"
+delta_cmd="${XDELTA_COMMAND:-xdelta3}"
+delta_decompress_flags="${XDELTA_ARGS:--d -s}"
+tar_cmd="${TAR_COMMAND:-tar}"
+tar_decompress_cmd="${tar_cmd} -xzvf"
 
 assert_requirements() {
-    local req_message
-    local command
-    local level
-    local COMMAND_NOT_FOUND_RC_THRESHOLD=100
-    local COMMAND_MISSING_ERROR_LEVEL="error"
+    local cmd
     local rc=0
-    local out_tmp
-
-    out_tmp=$(mktemp)
-    $app_sub_module REQS > "$out_tmp"
-    echo "${CONFIG_HAS_TAR_CMD}:error" >> "$out_tmp"
-    echo "${CONFIG_HAS_XDELTA3_CMD}:error" >> "$out_tmp"
     set +e
-    while IFS=":" read -r command level; do
-        "$command" < /dev/null > /dev/null 2>&1
-        if test $? -gt $COMMAND_NOT_FOUND_RC_THRESHOLD; then
-            echo "$level cannot find required command $command on PATH."
-            if test "$level" = "$COMMAND_MISSING_ERROR_LEVEL"; then
-                # only if the sub-module reported error level on absence of the command we bail out
-                rc=1
-            fi
-        fi
-    done < "$out_tmp"
-    set -e
-    rm -f "$out_tmp"
 
-    if test $rc -eq 1; then
-        # exit here is on purpose, as per the name of the function the update module will exit with an error
-        exit $rc
+    if ! $tar_cmd --version < /dev/null > /dev/null 2>&1; then
+        echo "error: cannot find required command ${tar_cmd} in PATH."
+        rc=127
+    fi
+    if ! $delta_cmd < /dev/null > /dev/null 2>&1; then
+        echo "warning: cannot find command '${delta_cmd}' in PATH."
     fi
-    return 0
+    $app_sub_module REQS
+    if [ $? -ne 0 ]; then
+        rc=1
+    fi
```

Comment:
there is no guarantee that command `[` will be present. that is why we use test everywhere here.